ema-examples/src/Ema/Example/Ex06_Markdown.hs

```diff
@@ -0,0 +1,170 @@
+    (pandoc, write) <- siteOutput (rp % _As @"Route") (pandocModel m) r
+    let head' = H.title "Basic site" >> H.base ! A.href "/"
+        body :: Text = coerce $ write pandoc
+        html = RU.renderHtml do
+          H.docType
+          H.html ! A.lang "en" $ do
+            H.head do
+              H.meta ! A.charset "UTF-8"
+              H.meta ! A.name "viewport" ! A.content "width=device-width, initial-scale=1"
+              head'
+            H.body $ H.preEscapedToHtml body
+    return $ AssetGenerated Html html
+
+wsConnDyn :: forall m. (MonadLoggerIO m) => Arg -> m (Dynamic m (TChan Route))
+wsConnDyn arg = do
+  value <- newTChanIO
+  let manage :: m ()
+      manage = do
+        logger <- askLoggerIO
+        let log = logger defaultLoc "wsConnDyn" LevelInfo
+        liftIO $ WS.runServer (editorWsAddress arg) (editorWsPort arg) \pendingConn -> do
+          conn :: WS.Connection <- WS.acceptRequest pendingConn
+          log "websocket connected"
+          WS.withPingThread conn 30 pass $
+            void $ infinitely do
+              msg <- liftIO $ toString @Text <$> WS.receiveData conn
+              log $ "got message: " <> show msg
+              baseDir <- makeAbsolute (Pandoc.argBaseDir $ pandocArg arg)
+              let fp = makeRelative baseDir msg
+              case Pandoc.mkPandocRoute fp of
+                Just (_, route)
+                  -- We should have received an absolute file path inside the base dir
+                  | isAbsolute msg && isRelative fp ->
+                      atomically $ writeTChan value (Route route)
+                _ -> pass
+  return $ Dynamic (value, const manage)
+
+main :: IO ()
+main = runWithFollow def
+
+runWithFollow ::
+  SiteArg Route ->
+  IO ()
+runWithFollow input = do
+  cli <- CLI.cliAction
+  result <- snd <$> runSiteWithServerOpts @Route followServerOptions cli input
+  case result of
+    CLI.Run _ :=> Identity () ->
+      flip runLoggerLoggingT (CLI.getLogger cli) $
+        CLI.crash "ema" "Live server unexpectedly stopped"
+    CLI.Generate _ :=> Identity _ -> pass
+
+followServerOptions :: EmaServerOptions Route
+followServerOptions = EmaServerOptions followClientJS followServerHandler
+
+followServerHandler :: EmaWsHandler Route
+followServerHandler conn model =
+  either id id <$> race (defaultEmaWsHandler @() conn ()) followHandler
+  where
+    rp = fromPrism_ $ routePrism model
+    log = logInfoNS "followServerHandler"
+    followHandler = do
+      route <- atomically $ readTChan $ wsNextRoute model
+      let Route pRoute = route
+          path = routeUrl rp route
+      if pRoute `member` Pandoc.modelPandocs (pandocModel model)
+        then do
+          log $ "switching to " <> show pRoute
+          liftIO $ WS.sendTextData conn $ "SWITCH " <> path
+        else log $ "invalid route " <> show pRoute
+      followHandler
+
+followJSShim :: Text
+followJSShim = decodeUtf8 $(embedFile "src/Ema/Example/Ex06_Markdown/ema-shim.js")
```

Comment:
The language server complains the file does not exist... I'm not sure what to do about this, it compiles in cabal. But the only difference from this ema shim to the default one is that it can read SWITCH "commands" from the server.

Reply:
We could also use the default shim and the REDIRECT command, but it would not use morphodom to switch

Reply:
Can't see this in the current PR anymore, so resolving ...